StringCatalogEnum/Tests/StringCatalogEnumTests/StringCatalogEnumTests.swift

```diff
@@ -50,5 +39,133 @@ final class StringKeyModelSpec: QuickSpec {
+                    "strings" : {
+                        "Home" : {
+                        },
+                    },
+                    "version" : "1.0"
+                }
+                """
+                guard let jsonData = json.data(using: .utf8) else {
+                    fatalError("Invalid JSON string")
+                }
+                let decoder = JSONDecoder()
+                expect{
+                    try decoder.decode(Localizations.self, from: jsonData)
+                }.toNot(throwError())
+
+                // Verify the decoded data
+                if let decodedData = try? decoder.decode(Localizations.self, from: jsonData) {
+                    // Verify the sourceLanguage
+                    expect(decodedData.sourceLanguage).to(equal("en"))
+
+                    // Verify the version
+                    expect(decodedData.version).to(equal("1.0"))
+
+                    // Verify the contents of 'strings' dictionary
+                    // Here we verify it with Not Be Nil because we design the model not to be optional
+                    expect(decodedData.strings["Home"]).toNot(beNil())
+                } else {
+                    fail("Failed to decode Localizations")
+                }
+            }
+
+            it("can decode the json data with English translation added") {
+                let json = """
+                {
+                    "sourceLanguage" : "en",
+                    "strings" : {
+                        "Login" : {
+                            "localizations" : {
+                                "en" : {
+                                    "stringUnit" : {
+                                        "state" : "translated",
+                                        "value" : "Login"
+                                    }
+                                }
+                            }
+                        },
+                    },
+                    "version" : "1.0"
+                }
+                """
+                guard let jsonData = json.data(using: .utf8) else {
+                    fatalError("Invalid JSON string")
+                }
+                let decoder = JSONDecoder()
+                expect{
+                    try decoder.decode(Localizations.self, from: jsonData)
+                }.toNot(throwError())
+
+                // Verify the decoded data
+                if let decodedData = try? decoder.decode(Localizations.self, from: jsonData) {
+                    // Verify the sourceLanguage
+                    expect(decodedData.sourceLanguage).to(equal("en"))
+
+                    // Verify the version
+                    expect(decodedData.version).to(equal("1.0"))
+
+                    // Verify the contents of 'strings' dictionary
+                    expect(decodedData.strings["Login"]).toNot(beNil())
+                    expect(decodedData.strings["Login"]?.localizations?["en"]?.stringUnit?.state).to(equal("translated"))
```

Comment:
Eventually we want to use `Decodable` for all these JSON objects, but this PR is good as is.